syntax = "proto3";

package product;

service ProductService{
    rpc CreateProduct (CreateProductRequest) returns (ProductResponse);

    rpc GetProduct (GetProductRequest) returns (ProductResponse);

    rpc UpdateProduct (UpdateProductRequest) returns (ProductResponse);

    rpc DeleteProduct (DeleteProductRequest) returns (DeleteProductResponse);

    rpc ListProducts(ListProductsRequest) returns (ListProductsResponse);

}

message CreateProductRequest
{
    string  StoreUuid = 1;
    string  Name = 2;
    bool    IsAvailable  = 3;
    float   Price = 4;
    string  CategoryUuid = 5;
    string  Description = 6;
    string  ImageUrl = 7;
}

message GetProductRequest 
{
    string ProductUuid = 1;  // Added ProductUuid for getting a specific product.
    string StoreUuid = 2;    // Keep StoreUuid in case you want to get a product in a specific store.

}

message ListProductsRequest
{
    string  StoreUuid = 1;
}


message DeleteProductRequest
{
    string  ProductUuid = 1;
}

message UpdateProductRequest
{
                string ProductUuid = 1;
    optional    string Name = 2;
    optional    bool IsAvailable  = 3;
    optional    float Price = 4;
    optional    string CategoryUuid = 5;
    optional    string Description = 6;
    optional    string ImageUrl = 7;
}

message ProductResponse
{
    bool    Success = 1 ;
    string  ProductUuid = 2;
    string  Name = 3;
    bool    IsAvailable  = 4;
    float   Price = 5;
    string  CategoryUuid = 6;
    string  Description = 7;
    string  ImageUrl = 8;
}

message DeleteProductResponse
{
    bool Success = 1;
}

message ListProductsResponse 
{
    repeated ProductResponse products = 1;
}



